docs/applications/development/clojure-deployment-with-wildfly-ubuntu-14-04.md

```diff
@@ -0,0 +1,241 @@
+
+3.  Update your system.
+
+        sudo apt-get update && sudo apt-get upgrade
+
+{: .note}
+>
+>This guide is written for a non-root user. Commands that require elevated privileges are prefixed with `sudo`. If you’re not familiar with the `sudo` command, you can check our [Users and Groups](/docs/tools-reference/linux-users-and-groups) guide.
+
+{: .note}
+>
+>In this guide `example.com` will be used as a domain name, and `linode-user` as a name of non-root user. Substitute your own FQDN and username accordingly.
+
+
+## Install Oracle JDK 8
+
+1.  Add Oracle Java 8 Installer PPA repository to the system:
+
+		sudo add-apt-repository ppa:webupd8team/java
+
+2.  Update package list and install Oracle JDK 8:
+
+		sudo apt-get update
+		sudo apt-get install oracle-java8-installer
+
+	You will be asked twice to agree with "Oracle Binary Code License Agreement for the Java SE Platform Products and JavaFX" by installer. Press &lt;Ok&gt; button to agree the first time, and &lt;Yes&gt; button the second time.
+
+3.  Make Oracle Java default:
+
+		sudo apt-get install oracle-java8-set-default
+
+4.  Check installation by running
+
+		java -version
+
+	Output should resemble the following:
+
+	    java version "1.8.0_66"
+		Java(TM) SE Runtime Environment (build 1.8.0_66-b17)
+		Java HotSpot(TM) 64-Bit Server VM (build 25.66-b17, mixed mode)
+
+{: .note}
+>
+>If you are not comfortable with using 3rd-party PPA, please use instructions for manual installation of Oracle Java 8 from [Java Development with WildFly on CentOS 7](docs/applications/development/java-development-wildfly-centos-7) guide.
+
+## Install Leiningen
+
+1. *Leiningen* is a Clojure project build tool. Install it system-wide:
+
+		sudo wget https://raw.githubusercontent.com/technomancy/leiningen/stable/bin/lein -O /usr/local/bin/lein
+		sudo chmod a+x /usr/local/bin/lein
+
+2.  To check installation has been successful, run
+
+		lein -v
+
+	Leiningen will update itself and then will output the version info, which should look similar to the following:
+
+		Leiningen 2.5.3 on Java 1.8.0_66 Java HotSpot(TM) 64-Bit Server VM
+
+## Create Sample Application
+
+Now you will create sample Clojure web application based on *Luminus* framework.
+
+1.  In your home directory run
+
+		lein new luminus clj-app
+
+	This will create new web application using Luminus framework application template. It will use `immutant` as a web server, which is the default option.
+
+2.  To check everything went smoothly, run newly created application in development mode:
+
+		cd clj-app
+		lein run
+
+	Now open http://&lt;public_linode_ip&gt;:3000/ in your browser and you will see the sample application main page.
+
+![Luminus application main page](/docs/assets/clj-luminus-main-page.png)
+
+	{: .note}
+	>
+	>Make sure port 3000 is open in firewall for this to work.
+
+3.  Stop the development server by pressing **Ctrl-C** in console.
+
+## Install JBoss WildFly Application Server
```

Comment:
I considered doing just that, but WildFly guide (https://www.linode.com/docs/applications/development/java-development-wildfly-centos-7) is CentOS specific and I feel like it would only complicate things. Also I think that installation script provided in Java WildFly guide is too complicated and WildFly installation could be done in a simpler way with a few explicit manual steps.